pkg/metrics/instance/group_manager.go

```diff
@@ -30,11 +37,152 @@ type GroupManager struct {
 
 	mtx sync.Mutex
 
+	// activeConfigs is a list of all the configs currently being used in the group
+	activeConfigs []Config
+
 	// groups is a map of group name to the grouped configs.
 	groups map[string]groupedConfigs
 
 	// groupLookup is a map of config name to group name.
 	groupLookup map[string]string
+
+	mergedConfigHashes map[string]int
+
+	log log.Logger
```

Comment:
nit: can you group these so there's no empty lines in between the members being protected by `mtx`? If you want a comment for a member, you can do it on the right:

```go
mtx sync.Mutex
activeConfigs []Config // List of individual configs currently applied
// etc
```